pkgs/development/tools/protoc-gen-rust/default.nix

```diff
@@ -0,0 +1,37 @@
+{ fetchFromGitHub
+, lib
+, rustPlatform
+, protobuf
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "protoc-gen-rust";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "stepancheg";
+    repo = "rust-protobuf";
+    sha256 = "sha256-oMcemWuhj9ZRAt93qEkgVghpNaC7TvRDHA9ExthS7Is=";
+    rev = "v${version}";
+  };
+
+  cargoSha256 = "sha256-5GTwngdytVM6OGjsj9xyGo7gesGr3y+cgcywTKP3obU=";
+
+  cargoBuildFlags = ["--bin" "protoc-gen-rust"];
+
+  # This patch is necessary as the repo doesn't provide a Cargo.lock
+  cargoPatches = [
+    ./add-Cargo-lock.patch
+  ];
+
+  nativeBuildInputs = [ protobuf ];
+  doCheck = true;
+
+  meta = with lib; {
+    description = "Protobuf plugin for generating Rust code";
+    homepage = "https://github.com/stepancheg/rust-protobuf";
+    license = licenses.mit;
+    maintainers = with maintainers; [ lucperkins ];
+    mainProgram = "dapr";
```

Comment:
copy-paste mistake? the binary is called `protoc-gen-rust`

Reply:
Ah, derp. Fixed!